src/pykeen/hpo/hpo.py

```diff
@@ -77,6 +76,7 @@ class Objective:
     loss_kwargs: Optional[Mapping[str, Any]] = None
     loss_kwargs_ranges: Optional[Mapping[str, Any]] = None
     # 4. Regularizer
+    regularizer: Optional[Type[Regularizer]] = None
```

Comment:
Or `Hint[Regularizer]`?

Reply:
hmmm i will check, that makes sense

Reply:
Nope. This needs to be pre-resolved. I actually changed the type hint, ran mypy, then checked the error to figure out that the code inside the Objective.__call__ doesn't do any resolving - it expects the regularizer class to be pre-resolved